source/game/kart/KartCollide.cc

```diff
@@ -360,6 +363,15 @@ void KartCollide::processWheel(CollisionData &collisionData, Hitbox &hitbox,
 void KartCollide::processBody(CollisionData &collisionData, Hitbox &hitbox,
         Field::CourseColMgr::CollisionInfo *colInfo, Field::KCLTypeMask *maskOut) {
     processFloor(collisionData, hitbox, colInfo, maskOut, false);
+    const auto colDirector = Field::CollisionDirector::Instance();
+    bool hitCannon =
+            colDirector->findClosestCollisionEntry(maskOut, KCL_TYPE_BIT(COL_TYPE_CANNON_TRIGGER));
+    if (hitCannon) {
```

Comment:
This can fit on one line
```suggestion
    if (colDirector->findClosestCollisionEntry(maskOut, KCL_TYPE_BIT(COL_TYPE_CANNON_TRIGGER))) {
```